python/pyspark/ml/param/__init__.py

```diff
@@ -52,10 +50,11 @@ class Params(Identifiable):
 
     __metaclass__ = ABCMeta
 
-    def __init__(self):
-        super(Params, self).__init__()
-        #: embedded param map
-        self.paramMap = {}
+    #: internal param map for user-supplied values param map
+    paramMap = {}
```

Comment:
That's the default behavior. `paramMap` doesn't need reference to `self`, so this is simpler.